spark-jobs/src/main/scala/filodb/repair/ChunkCopier.scala

```diff
@@ -0,0 +1,134 @@
+
+  val diskTimeToLiveSeconds = conf.getTimeAsSeconds("spark.filodb.chunkcopier.diskTimeToLive")
+  val splitsPerNode = conf.getInt("spark.filodb.chunkcopier.splitsPerNode", 1)
+  val batchSize = conf.getInt("spark.filodb.chunkcopier.batchSize", 10000)
+
+  val readSched = Scheduler.io("cass-read-sched")
+  val writeSched = Scheduler.io("cass-write-sched")
+
+  val sourceSession = FiloSessionProvider.openSession(sourceCassConfig)
+  val targetSession = FiloSessionProvider.openSession(targetCassConfig)
+
+  val sourceCassandraColStore = new CassandraColumnStore(sourceConfig, readSched, sourceSession)(writeSched)
+  val targetCassandraColStore = new CassandraColumnStore(targetConfig, readSched, targetSession)(writeSched)
+
+  private[repair] def getScanSplits = sourceCassandraColStore.getScanSplits(sourceDatasetRef, splitsPerNode)
+
+  def run(splitIter: Iterator[ScanSplit]): Unit = {
+    sourceCassandraColStore.copyChunksByIngestionTimeRange(
+      sourceDatasetRef,
+      splitIter,
+      ingestionTimeStart.toEpochMilli(),
+      ingestionTimeEnd.toEpochMilli(),
+      batchSize,
+      targetCassandraColStore,
+      targetDatasetRef,
+      diskTimeToLiveSeconds.toInt)
+  }
+
+  def shutdown(): Unit = {
+    sourceCassandraColStore.shutdown()
+    targetCassandraColStore.shutdown()
+  }
+}
+
+object ChunkCopier {
+  class ByteComparator extends java.util.Comparator[Array[Byte]] {
+    def compare(a: Array[Byte], b: Array[Byte]): Int = java.util.Arrays.compareUnsigned(a, b)
+  }
+
+  val cache = new java.util.TreeMap[Array[Byte], ChunkCopier](new ByteComparator)
+
+  // scalastyle: off null
+  def lookup(conf: SparkConf): ChunkCopier = synchronized {
+    // SparkConf cannot be used as a key, so serialize it instead.
+    val bout = new java.io.ByteArrayOutputStream()
+    val oout = new java.io.ObjectOutputStream(bout)
+    oout.writeObject(conf)
+    oout.close()
+    val key = bout.toByteArray()
+
+    var copier = cache.get(key)
+    if (copier == null) {
+      copier = new ChunkCopier(conf)
+      cache.put(key, copier)
+    }
+    copier
+  }
+  // scalastyle: on
+}
+
+/**
+  * For launching the Spark job.
+  */
+object ChunkCopierMain extends App with StrictLogging {
+  run(new SparkConf(loadDefaults = true))
+
+  def run(conf: SparkConf): SparkSession = {
+    logger.info(s"ChunkCopier Spark Job Properties: ${conf.toDebugString}")
+
+    val copier = ChunkCopier.lookup(conf)
+
+    val spark = SparkSession.builder()
+      .appName("FiloDBChunkCopier")
+      .config(conf)
+      .getOrCreate()
+
+    val splits = copier.getScanSplits
+
+    logger.info(s"Cassandra split size: ${splits.size}. We will have this many spark partitions. " +
+      s"Tune splitsPerNode which was $copier.splitsPerNode if parallelism is low")
+
+    spark.sparkContext
+      .makeRDD(splits)
+      .foreachPartition(splitIter => ChunkCopier.lookup(conf).run(splitIter))
```

Comment:
Did you verify that `conf` is same for each worker task ? For example, no additional properties are added/removed/updated ?

Reply:
Didn't check. Do you expect any to be added?

Reply:
The SparkConf docs state that the config cannot change after running.